crates/distribution-types/src/traits.rs

```diff
@@ -169,7 +169,7 @@ impl std::fmt::Display for Dist {
 
 impl std::fmt::Display for GitSourceDist {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}{}", self.name(), self.version_or_url())
+        write!(f, "{} @ git+{}", self.name(), self.url)
```

Comment:
This seems off to me... Why is `git+` not tracked in `version_or_url`?

Reply:
In `foo = { url = "https://github.com/foo/foo" }` we don't have a git prefix, while in `foo @ git+https://github.com/foo/foo`, we do. I went with the option to normalize both to no `git+` prefix, causing this oddity. (Adding the `git+` just for `.version_or_url()` doesn't work because it returns a `&'a VerbatimUrl`)

Reply:
Changed this back due to a different git fix